src/codegen/generate.ts

```diff
@@ -360,12 +368,26 @@ export default class ApiGenerator {
 
   /**
    * Create a type alias for the schema referenced by the given ReferenceObject
+   * @param ignoreDiscriminator If true, the discriminator property of the schema referenced by
+   * `obj` will be ignored. This is meant to be used when getting the type of a parent schema in an
+   * `allOf` constructor.
    */
-  getRefAlias(obj: OpenAPIV3.ReferenceObject, onlyMode?: OnlyMode) {
-    const { $ref } = obj;
+  getRefAlias(
+    obj: OpenAPIV3.ReferenceObject,
+    onlyMode?: OnlyMode,
+    ignoreDiscriminator?: boolean,
+  ) {
+    let { $ref } = obj;
+    if (ignoreDiscriminator) $ref = "ignoreDiscriminator:" + $ref;
```

Comment:
Could you explain your thinking behind the `"ignoreDiscriminator"` prefix? Where is it being picked up?

Reply:
Take the test I added as an example. We need to invent a unique reference for `PetBase` to take advantage of `refs` field. Plus, the reference must be different from "#/components/schemas/Pet". I could use "#/components/schemas/PetBase", but there might already be a schema named "PetBase". So, I've decided to use "ignoreDiscriminator:#/components/schemas/Pet".

Reply:
> Will we have both refs `ignoreDiscriminator:#/components/schemas/Pet` and `#/components/schemas/Pet` on the list?

Yes.

Reply:
Okay, I'll think about this. If you come up with a specific implementation, please send a PR to my branch anytime. I'll merge it :)

Reply:
Would love to get more into this but unfortunately need to balance my time in favor of other projects currently.

I was thinking that maybe this also is just a naming/understanding/clarity problem. 
A common pattern I notice in my head when reviewing is not understanding how all the parts are supposed to play together. 
Maybe we can up with a good name for the pattern _(unfortunately the Spec also just talks about the HOW, and not about the WHAT)_ and then align our internal naming & comments to that?

But again: I notice that I'm not really constructive here and I'm grateful for the work you put in, please take my comments as inspiration rather then requirements. We have tests, other tests aren't failing and that also means we can as well refactor the internals later once we learned more about it or run into actual issues.

Reply:
Another Thought: What if we actually go with `#/components/schemas/PetBase` and do a check beforehand if that name is actually free? (And if it isn't go with `#/components/schemas/PetBase2`?)

Reply:
> What if we actually go with `#/components/schemas/PetBase` and do a check beforehand if that name is actually free? (And if it isn't go with `#/components/schemas/PetBase2`?)

That sounds natural to me. I'll give it a try.